sensor/common/signal/component/component.go

```diff
@@ -0,0 +1,143 @@
+package component
+
+import (
+	"io"
+	"strings"
+
+	"github.com/stackrox/rox/generated/internalapi/central"
+	"github.com/stackrox/rox/generated/internalapi/sensor"
+	"github.com/stackrox/rox/pkg/centralsensor"
+	"github.com/stackrox/rox/pkg/concurrency"
+	"github.com/stackrox/rox/pkg/logging"
+	"github.com/stackrox/rox/pkg/stringutils"
+	"github.com/stackrox/rox/sensor/common"
+	"github.com/stackrox/rox/sensor/common/message"
+)
+
+const maxBufferSize = 10000
+
+var (
+	log = logging.LoggerForModule()
+)
+
+type Component interface {
+	common.SensorComponent
+
+	GetReceiver() chan *sensor.ProcessSignal
+}
+
+type componentImpl struct {
+	common.SensorComponent
+
+	processPipeline Pipeline
+	indicators      chan *message.ExpiringMessage
+	receiver        chan *sensor.ProcessSignal
+	writer          io.Writer
+
+	stopper concurrency.Stopper
+}
+
+type Option func(*componentImpl)
+
+// WithTraceWriter sets a trace writer that will write the messages received from collector.
+func WithTraceWriter(writer io.Writer) Option {
+	return func(cmp *componentImpl) {
+		cmp.writer = writer
+	}
+}
+
+func New(pipeline Pipeline, indicators chan *message.ExpiringMessage, opts ...Option) Component {
+	cmp := &componentImpl{
+		processPipeline: pipeline,
+		indicators:      indicators,
+		receiver:        make(chan *sensor.ProcessSignal, maxBufferSize),
```

Comment:
Why do we need the buffer here? 
It looks like we are only reading from this channel, so I am wondering what is the purpose of this channel?